openmls/src/group/mls_group/creation.rs

```diff
@@ -21,11 +20,11 @@ impl MlsGroup {
     ///
     /// Returns an error ([`NewGroupError::NoMatchingCredentialBundle`]) if the
     /// private key for the [`SignaturePublicKey`] can not be found.
-    pub fn new(
-        backend: &impl OpenMlsCryptoProvider,
+    pub fn new<KeyStore: OpenMlsKeyStore>(
+        backend: &impl OpenMlsCryptoProvider<KeyStoreProvider = KeyStore>,
         mls_group_config: &MlsGroupConfig,
         signature_key: &SignaturePublicKey,
-    ) -> Result<Self, NewGroupError> {
+    ) -> Result<Self, NewGroupError<KeyStore::Error>> {
```

Comment:
Another way of writing these would be the following, fully moving to generics.

```rust
pub fn new<T: OpenMlsCryptoProvider>(
        backend: &T,
        mls_group_config: &MlsGroupConfig,
        signature_key: &SignaturePublicKey,
    ) -> Result<Self, NewGroupError<<<T as openmls_traits::OpenMlsCryptoProvider>::KeyStoreProvider as OpenMlsKeyStore>::Error>>{
```

Reply:
True, that would work as well. Would you prefer that style? I have no particular preference here.

Reply:
I'd personally prefer the one that's in the code right now. If you don't have a strong opinion, I'd prefer to leave it as it is.